src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -374,7 +374,21 @@ def reflect_account_groups_on_workspace(self):
             raise ManyError(errors)
 
     def get_migration_state(self) -> MigrationState:
-        return MigrationState(self.snapshot())
+        state = MigrationState(self.snapshot())
```

Comment:
I am not sure this is the best way to solve the issue. It would be better to filter when we fetch the groups (`_fetcher` method). We are already filtering the groups when we crawl. We just need to add simple filter when we fetch as well. This would be one-liner fix.